### PR TITLE
Remove temporary pickle files

### LINKED ISSUE #1353: `_pickle_and_log_artifact` doesn't clean up pickle file after logging it as an artifact
This line doesn't seem to clean up the pickle file after logging it as an mlflow artifact.

https://github.com/microsoft/FLAML/blob/c90946f3035d5886044e6f9cb30400e3a645434b/flaml/fabric/mlflow.py#L363

This broke our CI (which doesn't allow untracked files being create during test):

https://github.com/mlflow/mlflow/actions/runs/10933548130/job/30352200311

```
Untracked files found:
feature_transformer.pkl
label_transformer.pkl
```

### PR DESCRIPTION
**Pull request by thinkall**
## Why are these changes needed?

To fix #1353

## Related issue number

Closes #1353

## Checks

- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: flaml/fabric/mlflow.py [hunk 1]
```diff
@@ -354,6 +354,7 @@ def _pickle_and_log_artifact(self, obj, artifact_name, pickle_fpath="temp_.pkl")
         with open(pickle_fpath, "wb") as f:
             pickle.dump(obj, f)
         mlflow.log_artifact(pickle_fpath, artifact_name)
+        os.remove(pickle_fpath)
```
Comment (harupy) [inline flaml/fabric/mlflow.py:357]: what if `log_artifact` fails?

Comment (harupy) [reply flaml/fabric/mlflow.py:357]: the code would look like

```python
with tempfile.TemporaryDirectory() as tmpdir:
    pickle_path = tmpdir/path
    # save pickle
    mlflow.log_artifact(pickle_path)

# pickle file deleted when exiting the context manager even if log_artifact fails
```

Comment (harupy) [reply flaml/fabric/mlflow.py:357]: Creating files in the current working directory is not preferred. Imagine you're working in a code editor, and a pickle file suddenly shows up and disappears.

Comment (thinkall) [reply flaml/fabric/mlflow.py]: @harupy,  thanks for the suggestions, updated.

**Review by leafy-lee — APPROVED** [review]

**Pull request merged**